File: utils/dnp3_src/dotnet/bindings/CLRInterface/config/MasterConfig.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Serialization;

namespace Automatak.DNP3.Interface
{
    /// <summary>
    /// Configuration information for the dnp3 master
    /// </summary>
    [Serializable]
    public class MasterConfig
    {

        /// <summary>
        /// Constructor with reasonable defaults
        /// </summary>
        public MasterConfig()
        {            
            timeSyncMode = TimeSyncMode.None;
            disableUnsolOnStartup = true;
            unsolClassMask = ClassField.AllEventClasses;
            startupIntegrityClassMask = ClassField.AllClasses;
            integrityOnEventOverflowIIN = true;
            eventScanOnEventsAvailableClassMask = ClassField.None;
            responseTimeout = TimeSpan.FromSeconds(5);
            taskRetryPeriod = TimeSpan.FromSeconds(5);
            taskStartTimeout = TimeSpan.FromSeconds(10);
        }
        
        /// <summary>
        /// If true, the master will do time syncs when it sees the time IIN bit from the outstation
        /// </summary>
        public TimeSyncMode timeSyncMode;

        /// <summary>
        /// If true, the master will disable unsol on startup for all 3 classes
        /// </summary>
        public bool disableUnsolOnStartup;

        /// <summary>
        /// Bitwise mask used determine which classes are enabled for unsol, if 0 unsol is not enabled
        /// </summary>
        public ClassField unsolClassMask;

        /// <summary>
        /// Which classes should be requested in a startup integrity scan, defaults to 3/2/1/0
        /// A mask equal to 0 means no startup integrity scan will be performed
        /// </summary>
        public ClassField startupIntegrityClassMask;

        /// <summary>
        /// Defines whether an integrity scan will be performed when the EventBufferOverflow IIN is detected
        /// </summary>
        public bool integrityOnEventOverflowIIN;

        /// <summary>
        /// Which classes should be requested in an event scan when detecting corresponding events available IIN
        /// </summary>
        public ClassField eventScanOnEventsAvailableClassMask;

        /// <summary>
        /// Application layer response timeout
        /// </summary>
        [XmlIgnore]
        public TimeSpan responseTimeout;

        /// <summary>
        /// Time delay beforce retrying a failed task
        /// </summary>
        [XmlIgnore]
        public TimeSpan taskRetryPeriod;

        /// <summary>
        /// Time delay beforce retrying a failed task
        /// </summary>
        [XmlIgnore]
        public TimeSpan taskStartTimeout;


        [XmlElement]
        public long TaskResponseTimeoutMilliseconds
        {
            get
            {
                return (responseTimeout.Ticks / TimeSpan.TicksPerMillisecond);
            }
            set
            {
                responseTimeout = TimeSpan.FromMilliseconds(value);
            }
        }
        

        [XmlElement]
        public long TaskRetryPeriodMilliseconds
        {
            get
            {
                return (taskRetryPeriod.Ticks / TimeSpan.TicksPerMillisecond);
            }
            set
            {
                taskRetryPeriod = TimeSpan.FromMilliseconds(value);
            }
        }

        [XmlElement]
        public long TaskStartTimeoutMilliseconds
        {
            get
            {
                return (taskStartTimeout.Ticks / TimeSpan.TicksPerMillisecond);
            }
            set
            {
                taskStartTimeout = TimeSpan.FromMilliseconds(value);
            }
        }
    }

}

File: utils/dnp3_src/java/bindings/src/main/java/com/automatak/dnp3/MasterConfig.java
/**
 * Copyright 2013-2016 Automatak, LLC
 *
 * Licensed to Automatak, LLC (www.automatak.com) under one or more
 * contributor license agreements. See the NOTICE file distributed with this
 * work for additional information regarding copyright ownership. Automatak, LLC
 * licenses this file to you under the Apache License Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain
 * a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0.html
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 */
package com.automatak.dnp3;

import com.automatak.dnp3.enums.TimeSyncMode;

import java.time.Duration;

/**
 * Configuration class for a master station
 */
public class MasterConfig {


    /**
     * Constructor with reasonable defaults
     */
    public MasterConfig()
    {

    }

    /**
     * Application layer response timeout
     */
    public Duration responseTimeout = Duration.ofSeconds(5);

    /**
     * Enumeration that specifies how/if the master does performs time synchronization
     */
    public TimeSyncMode timeSyncMode = TimeSyncMode.None;

    /**
     * If true, the master will disable unsol on startup for all 3 event classes
     */
    public boolean disableUnsolOnStartup = true;

    /**
     * If true, the master will not clear the restart IIN bit in response to detecting it set
     */
    public boolean ignoreRestartIIN = false;

    /**
     * If true, the master will enable/disable unsol on startup
     */
    public ClassField unsolClassMask = ClassField.allEventClasses();

    /**
     * Which classes should be requested in a startup integrity scan, defaults to 3/2/1/0
     * A mask equal to 0 means no startup integrity scan will be performed
     */
    public ClassField startupIntegrityClassMask = ClassField.allClasses();

    /**
     * Defines whether an integrity scan will be performed when the EventBufferOverflow IIN is detected
     */
    public boolean integrityOnEventOverflowIIN = true;

    /**
     * Which classes should be requested in an event scan when detecting corresponding events available IIN
     */
    public ClassField eventScanOnEventsAvailableClassMask = ClassField.none();

    /**
     * Time delay beforce retrying a failed task
     */
    public Duration taskRetryPeriod = Duration.ofSeconds(5);

    /**
     * Period after which tasks that haven't started are failed
     */
    public Duration taskStartTimeout = Duration.ofSeconds(10);

    /**
     * maximum APDU tx size in bytes
     */
    public int maxTxFragSize = 2048;

    /**
     * maximum APDU rx size in bytes
     */
    public int maxRxFragSize = 2048;
}
